pkg/controller/subscription/subscription_controller.go

```diff
@@ -73,13 +83,74 @@ func (r *ReconcileSubscription) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	if instance.Spec.InstallPlanApproval != v1alpha1.ApprovalManual {
-		instance.Spec.InstallPlanApproval = v1alpha1.ApprovalManual
-		err = r.client.Update(context.TODO(), instance)
+	if subscription.Spec.InstallPlanApproval != operatorsv1alpha1.ApprovalManual {
+		subscription.Spec.InstallPlanApproval = operatorsv1alpha1.ApprovalManual
+		err = r.client.Update(context.TODO(), subscription)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
-	return reconcile.Result{}, nil
+	result, err := r.HandleUpgrades(context.TODO(), subscription)
+
+	return result, err
+}
+
+func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscription *operatorsv1alpha1.Subscription) (reconcile.Result, error) {
+	if !rhmiConfigs.IsUpgradeAvailable(rhmiSubscription) {
+		logrus.Infof("no upgrade available")
+		return reconcile.Result{}, nil
+	}
+
+	logrus.Infof("RHMI upgrade available")
+
+	latestRHMIInstallPlan, err := rhmiConfigs.GetLatestInstallPlan(ctx, rhmiSubscription, r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	latestRHMICSV, err := rhmiConfigs.GetCSV(latestRHMIInstallPlan)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	config := &integreatlyv1alpha1.RHMIConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      installation.DefaultCloudResourceConfigName,
```

Comment:
Should this not be `rhmi-config`? :thinking: 
https://github.com/integr8ly/integreatly-operator/blob/8b6d1aa5157a7135bd7353a7f99814fff15db2b4/pkg/controller/installation/bootstrapReconciler.go#L129

`cloud-resource-config` is the name of the cloud resource config map for forcing whether to use in cluster or aws storage I believe

Reply:
oh, this is strange, is `DefaultCloudResourceConfigName` used anywhere then?

Reply:
oh, it's the name of the configmap for CRO :facepalm: I'll update now